test/Microsoft.TestPlatform.TestUtilities/RandomId.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Security.Cryptography;
+
+namespace Microsoft.TestPlatform.TestUtilities;
+
+/// <summary>
+/// Slightly random id that is just good enough for creating disctinct directories for each test.
+/// </summary>
+public static class RandomId
```

Comment:
Using this over guid just makes the output easier to read when there are 5 occurences of the random id. WE could also use it for our real log once we have templating and have <id> (short id) and <guid> (long id).